src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -557,38 +556,131 @@ internal static class HashAlgorithm
         {
             private const string TypeName = "System.Security.Cryptography.HashAlgorithm";
 
-            internal static readonly Type Type = ReflectionUtilities.GetTypeFromEither(
+            private static Type s_lazyType;
+
```

Comment:
A couple of benefits:
1) No worries about order of initialization when the reflected types and members are referenced one from another.
2) A bunch of these APIs are only used on certain platforms, so there is no point in wasting startup time to initialize them.